### ISSUE TITLE
rwkv7 comments per request

### ISSUE THREAD

**Issue by SmerkyG** [RWKV/RWKV-block#1]
After reviewing the rwkv7 blocks, here are some comments and requested changes:

- [x] LoRA sizes should be read from config, and default to the suggestion value. See HF code for my proposed naming of these.
- [x] self.v0,v1,v2 should not exist on layer zero (if layer_id==0 skip creating these)
- [x] probably better to make `eps=1e-5*head_size` rather than `eps=(1e-5)*(head_size_divisor**2)` for model scalability reasons... 
- [x] `v_first_state` should just be `v_first` - it's not a state held across iterations, and should overwrite the existing v_first at that point
- [x] there are a lot of different v7 attn implementations, with different algorithms and different tolerances for larger headsize, and also different known precisions, not sure which you want to support
- [x] what benefit does `load_from_model_state_dict` confer? it seems exactly identical to torch.nn.Module.load_state_dict but with fewer options
- [x] https://github.com/RWKV/RWKV-block/blob/bd286a12a934429837b862a7ac293a43513dfafc/rwkv/v7_goose/model/rwkv7_goose_model.py#L147 "compute the x embedding" is very nonstandard terminology - "compute the hidden state" would be better
- [x] `layer_id = cMap.get_layer_id(-1)` the cMap should be immutable, and you can pass in a layer_id
- [x] naming convention would be better to not use 'goose' to mean only the full model, since rwkv7==goose. So maybe `rwkv7_model_config_map.py` and `rwkv7_model.py`
- [x] I'm not sure having the layer_id be part of the config map is useful - I've never seen any other library pass it that way
- [x] same for device and dtype not being in the cMap... typically this is set in the calling code by using a contextmanager, so that the model code does not have to set these manually for each tensor
- [x] initializations... this is a tricky one, but basically it's inefficient to have these be in the actual `__init__` functions - they should ideally be in a different function such as reset_parameters(). Better to initialize everything as empty tensors and then allow the user to call `reset_parameters()` if they want inits. Otherwise for large models you end up taking a long time initializing these and just overwriting them immediately on load. Especially true for DS3/FSDP
see accepted solution in https://discuss.pytorch.org/t/how-to-re-set-alll-parameters-in-a-network/20819/12 for some related code
- [x] 'cMap' is not a great name to mean normalized 'configMap' - if you want to replace it, both names should not exist at the same time so better to call it cMap or configMap the whole way through
